scripts/corelli/calibration/utils.py

```diff
@@ -75,6 +78,57 @@ def bank_numbers(bank_selection: str) -> List[str]:
     return [str(n) for n in sorted(banks)]
 
 
+def preprocess_banks(input_workspace: str, output_workspace: str) -> Workspace2D:
+    r"""
+    Clone the input workspace/run and preprocess the histograms for each tube such
+    that the peak finding algorithm can have a better chance of finding the correct
+    wire position.
+
+    Note: the preprocess step needs to be performed after load_banks, which provided
+    the bank selection.
+    :param input_workspace: input workspace name
+    :param output_workspace: output workspace with pre-processed historgrams
+    """
+    CloneWorkspace(InputWorkspace=str(input_workspace), OutputWorkspace=output_workspace)
```

Comment:
What if the user wants to overwrite the input workspace? After all, the input workspace in the output of `load_banks`, with each pixel containing the integrated intensity.
We could have optional `output_workspace=None` and in this case, overwrite `input_workspace`

Reply:
The outcome of this function is __only__ suitable for calibration as the histograms are modified to assist the subsequent identification of wire shadow positions.  Therefore, I have to insist on having the function to cowardly refuse any attempt of in-place operation of `input_workspace`.